nitro-protocol/contracts/NitroAdjudicator.sol

```diff
@@ -93,19 +95,21 @@ contract NitroAdjudicator is ForceMove, MultiAssetHolder {
      * @notice Encodes application-specific rules for a particular ForceMove-compliant state channel.
      * @dev Encodes application-specific rules for a particular ForceMove-compliant state channel.
      * @param fixedPart Fixed part of the state channel.
-     * @param signedVariableParts Array of variable parts to find the latest of.
-     * @return VariablePart Latest supported by application variable part from supplied array.
+     * @param proof Variable parts of the states with signatures in the support proof. The proof is a validation for the supplied candidate.
+     * @param candidate Variable part of the state to change to. The candidate state is supported by proof states.
      */
-    function latestSupportedState(
+    function requireStateSupported(
         FixedPart calldata fixedPart,
-        SignedVariablePart[] calldata signedVariableParts
+        SignedVariablePart[] calldata proof,
+        SignedVariablePart calldata candidate
     ) external pure returns (VariablePart memory) {
         // To avoid `Stack to deep` error, signedVariableParts are copied to `memory` array explicitly
-        SignedVariablePart[] memory _signedVariableParts = signedVariableParts;
+        SignedVariablePart[] memory _proof = proof;
```

Comment:
Out of scope for this PR: I tried removing this manual step and everything still worked :-)

Reply:
[bfb398d](https://github.com/statechannels/go-nitro/pull/817/commits/bfb398deefec7988c0a7c0665dac902475f3743d)

Reply:
Wow, solidity surprises me every now and then!